Composer/packages/extensions/intellisense/src/components/CompletionElement.tsx

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { CompletionItemKind } from 'monaco-languageclient';
+import { FontIcon } from 'office-ui-fabric-react/lib/Icon';
+import { DirectionalHint, TooltipHost } from 'office-ui-fabric-react/lib/Tooltip';
+import React, { CSSProperties } from 'react';
+import { CompletionItem, MarkupContent } from 'vscode-languageserver-types';
+
+type FuseJsMatch = { indices: number[][]; value: string; key: string };
+
+const styles: Record<string, CSSProperties> = {
+  completionElement: {
+    height: '32px',
+    cursor: 'pointer',
+    padding: '0 4px',
+    width: '100%',
+    display: 'flex',
+    alignItems: 'center',
+  },
+  selectedElement: { backgroundColor: '#ddd' },
+  text: { fontSize: '15px' },
+  icon: { marginRight: '5px' },
+};
+
+const renderIcon = (kind: CompletionItemKind | undefined): JSX.Element => {
+  let icon: JSX.Element = <> </>;
```

Comment:
Nit: You don't have to declare a mutable let variable here if you put a default: in your switch

Reply:
Actually all this function is doing is figuring out the iconName, and passing it to FontIcon, so DRY?

Reply:
Good points. I refactored to address both your comments.